nix/dynect-record.nix

```diff
@@ -0,0 +1,55 @@
+{ config, pkgs, lib, uuid, name, ...}:
+with lib;
+with (import ./lib.nix lib);
+let
+  aRecordSubModule =
+  { options, ... }:
+  { options = {
+      address = mkOption {
+        type = types.str;
+        description = "IPv4 Address.";
+        example = "127.0.0.1";
+      };
+    };
+  };
+in
+{
+  options = {
+    name = mkOption {
+      default = "charon-${uuid}-${name}";
```

Comment:
Please change `charon` to `nixops`. (There are a few places where we use `charon` for backward compatibility, but obviously that's not needed for new resource types.)